src/reconcile/rec_visibility.c

```diff
@@ -531,6 +531,10 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *first_upd
         if ((txnid = upd->txnid) == WT_TXN_ABORTED)
             continue;
 
+        /* Give up if the update is from this transaction and on the metadata file.*/
```

Comment:
I had made that change (which triggered update restores) but it then had follow on failures. This was the safest change but happy to revisit it.

Reply:
Hmm, ok. Maybe somewhere else we are not checking for the same transaction ID. I am fine with the change, as it just fails the reconciliation instead of doing an update restore eviction.